engine/src/main/java/org/entando/entando/web/page/PageController.java

```diff
@@ -189,15 +187,42 @@ public void setAuthorizationService(PageAuthorizationService authorizationServic
             throw new ValidationGenericException(bindingResult);
         }
 
+        PagePositionRequest pagePositionRequest = new PagePositionRequest();
+        pagePositionRequest.setParentCode(pageRequest.getParentCode());
+        pagePositionRequest.setCode(pageCode);
+        int position = pageService.getPages(pageCode).size() + 1;
+        pagePositionRequest.setPosition(position);
+
+        validateMovePage(pageCode, bindingResult, pagePositionRequest);
+
         PageDto page = this.getPageService().updatePage(pageCode, pageRequest);
         Map<String, String> metadata = new HashMap<>();
         return new ResponseEntity<>(new RestResponse<>(page, metadata), HttpStatus.OK);
     }
 
+    private void validateMovePage(@PathVariable String pageCode,
```

Comment:
Is this the right place for this? Where does this same logic occur when we move a page on the main page tree? Can we call that instead?

Reply:
This is a small refactoring, It's the old main page tree code moved to this new method (to avoid code duplication) and reused here.